Upgrade samples to use Traefik 2.0 (#5)
configit-samples/clm-hosting: README.md

```yaml
services:
  # The "storage" service covers the single Storage API node.
  storage:
    loadBalancer:
      # A health check is added, allowing Traefik to detect if a server is down.
      healthCheck:
        path: /health
        interval: "30s"
        timeout: "3s"
      servers:
        - url: http://localhost:9021

  # The "configuration" service corresponds to a single Configuration API node.
  configuration:
    loadBalancer:
      # A health check is added, allowing Traefik to detect if a server is down.
      healthCheck:
        path: /health
        interval: "30s"
        timeout: "3s"
      servers:
        - url: http://localhost:9011
        - url: http://localhost:9012
        - url: http://localhost:9013

  # The "ui" service for the configurator web app and documentation.
  ui:
    loadBalancer:
      # A health check is added, allowing Traefik to detect if a server is down.
      healthCheck:
        path: /health
        interval: "30s"
        timeout: "3s"
      servers:
        - url: http://localhost:9001
```

This change adds the following:

- The Configuration API is scaled to three nodes, providing failover.

> - Traefik queries the `/health` endpoint every 30 seconds, removing nodes that are down. When node is up again, Traefik will discover it and start directing requests to the node again.

Reviewer: ```suggestion
- Traefik queries the `/health` endpoint every 30 seconds, removing nodes that are down. When a node is up again, Traefik will discover it and start directing requests to the node again.
```

In the redundancy example above, we scaled the Configuration API to three nodes.

> We did not configure any load balancing settings and in practice, this means that Traefik will use a round-robin strategy, distributing requests evenly on the three nodes.

Reviewer: ```suggestion
We did not configure any load balancing settings and, in practice, this means that Traefik will use a round-robin strategy, distributing requests evenly on the three nodes.
```

We can modify the `configuraton` section:

```yaml

> # The "configuration" service is a weighted round robin load-balancer based on three nodes.

Reviewer: ```suggestion
# The "configuration" service is a weighted round-robin load-balancer based on three nodes.
```

Reviewer: (tmi: "load balancer" is a noun here. when used as an adjective, it is hyphenated.)

> # The nodes are divided into two groups with weights of 4 and 2 respectively.

Reviewer: ```suggestion
# The nodes are divided into two groups with weights of 4 and 1 respectively.
```

configuration:
  weighted:
    services:
      - name: configuration-powerful
        weight: 4
      - name: configuration-backup

>         weight: 1

Reviewer: Is it okay that this sample doesn't exactly match the 3_lb_round_robin.yaml (which has a weight 2)?

Author: No, I think they should match 👍 Will fix it.

configuration-powerful:
  loadBalancer:
    healthCheck:
      path: /health
      interval: "30s"
      timeout: "3s"
    servers:
      - url: http://localhost:9011
      - url: http://localhost:9012
configuration-backup:
  loadBalancer:
    healthCheck:
      path: /health
      interval: "30s"
      timeout: "3s"
    servers:
      - url: http://localhost:9013
```

This change adds the following:

- We use a weighted load-balancing strategy and divide the nodes into two services.
- We assign a `weight` to the services. The weight describes how well the services are expected to perform relative to other services.

>   For example, a service with the weight 2 should be twice as fast as a service with a weight of 1.

Reviewer: ```suggestion
  For example, a service with the weight 4 should be four times as fast as a service with a weight of 1.
```
